cli/cmd/inject.go

```diff
@@ -196,8 +199,129 @@ func injectPodTemplateSpec(t *v1.PodTemplateSpec, controlPlaneDNSNameOverride, v
 	return true
 }
 
+func injectResource(bytes []byte, version string) ([]byte, error) {
```

Comment:
mind arranging this function below `InjectYAML`, so as to make the diff as small as possible? it's hard to tell if anything's changed otherwise.

if you're intentionally arranging private functions above public ones, that's cool, just consider doing it in a separate commit.

Reply:
Doesn't look like we were so lucky. 

```go

	case "List":
		var sourceList v1.List
		if err := yaml.Unmarshal(bytes, &sourceList); err != nil {
			return nil, err
		}

		items := []runtime.RawExtension{}

		for _, item := range sourceList.Items {
			resource, err := injectResource(item.Raw, version)
			if err != nil {
				return nil, err
			}

			items = append(items, runtime.RawExtension{Raw: resource})
		}

		obj = &v1.List{Items: items}
```

Is 99% of what changed. I moved a couple `if err`s into oneliners and left everything else as intact as possible.